examples/binarySearch/binarySearch.act

```diff
@@ -0,0 +1,44 @@
+behaviour init of BinarySearch
+interface constructor()
+
+creates
+	uint256[] array := []
+
+invariants
+	sorted(array)
+
+where
+
+sorted : uint256[] -> bool
+sorted([]) = true
+sorted([x]) = true
+sorted(x::y::xs) = x <= y and sorted(y::xs)
+
+
+// Returns the index of elem if elem in array,
+// otherwise returns -1.
+behaviour search of BinarySearch
+interface search(uint256 elem)
+
+returns binarySearch(array, elem, 0, array.length)
+
+ensures
+
+if elem in array
+then array[index] = elem
+else index = -1
+
+where
+
+index = binarySearch(array, elem, 0, array.length)
+
+binarySearch(uint256[] a, uint256 elem, uint256 left, uint256 right) =
```

Comment:
```suggestion
binarySearch(uint256[] a, uint256 elem, uint256 left, uint256 right) :=
```

Reply:
`sorted` uses `=`, so `=` or `:=`?

Reply:
let's go with `:=` for now. We are using `:=` for assignment in storage declarations right now so this would be in line with that

Reply:
updated